src/main/java/com/bakdata/kafka/ErrorLoggingFlatTransformer.java

```diff
@@ -0,0 +1,165 @@
+            final @NonNull Transformer<? super K, ? super V, ? extends Iterable<R>> transformer,
+            final @NonNull Predicate<Exception> errorFilter) {
+        return new ErrorLoggingFlatTransformer<>(transformer, errorFilter);
+    }
+
+    /**
+     * Wrap a {@code TransformerSupplier} and log thrown exceptions with input key and value. Recoverable Kafka
+     * exceptions such as a schema registry timeout are forwarded and not captured.
+     *
+     * @param supplier {@code TransformerSupplier} whose exceptions should be logged
+     * @param <K> type of input keys
+     * @param <V> type of input values
+     * @param <R> type of map result
+     * @return {@code TransformerSupplier}
+     * @see #logErrors(TransformerSupplier, Predicate)
+     * @see ErrorUtil#isRecoverable(Exception)
+     */
+    public static <K, V, R> TransformerSupplier<K, V, Iterable<R>> logErrors(
+            final @NonNull TransformerSupplier<? super K, ? super V, ? extends Iterable<R>> supplier) {
+        return logErrors(supplier, ErrorUtil::isRecoverable);
+    }
+
+    /**
+     * Wrap a {@code TransformerSupplier} and log thrown exceptions with input key and value.
+     * <pre>{@code
+     * final TransformerSupplier<K, V, Iterable<KeyValue<KR, VR>>> transformer = ...;
+     * final KStream<K, V> input = ...;
+     * final KStream<KR, VR> output = input.flatTransform(logErrors(transformer));
+     * }
+     * </pre>
+     *
+     * @param supplier {@code TransformerSupplier} whose exceptions should be logged
+     * @param errorFilter expression that filters errors which should be thrown and not logged
+     * @param <K> type of input keys
+     * @param <V> type of input values
+     * @param <R> type of map result
+     * @return {@code TransformerSupplier}
+     */
+    public static <K, V, R> TransformerSupplier<K, V, Iterable<R>> logErrors(
+            final @NonNull TransformerSupplier<? super K, ? super V, ? extends Iterable<R>> supplier,
+            final @NonNull Predicate<Exception> errorFilter) {
+        return new TransformerSupplier<>() {
+            @Override
+            public Set<StoreBuilder<?>> stores() {
+                return supplier.stores();
+            }
+
+            @Override
+            public Transformer<K, V, Iterable<R>> get() {
+                return logErrors(supplier.get(), errorFilter);
+            }
+        };
+    }
+
+    @Override
+    public void close() {
+        this.wrapped.close();
+    }
+
+    @Override
+    public void init(final ProcessorContext context) {
+        this.wrapped.init(context);
+    }
+
+    @Override
+    public Iterable<R> transform(final K key, final V value) {
+        try {
+            return this.wrapped.transform(key, value);
+        } catch (final Exception e) {
+            if (this.errorFilter.test(e)) {
+                throw e;
+            }
+            log.error("Cannot process ('" + ErrorUtil.toString(key) + "', '" + ErrorUtil.toString(value) + "')", e);
```

Comment:
You can use Slf4j's formatting string here, see https://www.slf4j.org/faq.html#paramException